stac_api/infrastructure/construct.py

```diff
@@ -71,12 +71,23 @@ def __init__(
         )
 
         lambda_function.add_environment(
-            "VEDA_STAC_PATH_PREFIX", veda_stac_settings.path_prefix
+            "VEDA_STAC_PATH_PREFIX", veda_stac_settings.stac_path_prefix
         )
 
         stac_api_integration = (
             aws_apigatewayv2_integrations_alpha.HttpLambdaIntegration(
-                construct_id, handler=lambda_function
+                construct_id,
+                handler=lambda_function,
+                parameter_mapping=(
+                    aws_apigatewayv2_alpha.ParameterMapping().overwrite_header(
+                        "host",
+                        aws_apigatewayv2_alpha.MappingValue(
+                            veda_stac_settings.domain_hosted_zone_name
+                        )
+                        if veda_stac_settings.domain_hosted_zone_name
+                        else None,
+                    )
+                ),
             )
```

Comment:
this is awesome + I wonder if the prefix can be managed here also (in the future)?

Question: I need to try this out but to me it seems that if domain_hosted_zone_name is None, then "host" will be overwritten with None? Or does it just end up being a no-op?

Reply:
I wanted to make it like a no-op if hosted zone wasn't present. I can double check if that's what it's doing